class-components/src/app/lib/utils.test.ts

```diff
@@ -0,0 +1,32 @@
+import { describe, it, expect } from 'vitest';
+import { checkPasswordStrength } from './utils';
+
+describe('checkPasswordStrength', () => {
```

Comment:
would this work?

```suggestion
describe('checkPasswordStrength', () => {
  it.each([
    ['', 0, 'empty string'],
    ['password', 1, 'only lowercase'],
    ['Password', 2, 'lowercase and uppercase'],
    ['Password123', 3, 'lowercase, uppercase, and number'],
    ['Password123!', 4, 'all four criteria'],
    ['Password!', 3, 'missing number'],
    ['!@#$%^&*', 1, 'only special characters'],
  ])('should return %d for a password with %s', (password, expectedScore, description) => {
    expect(checkPasswordStrength(password)).toBe(expectedScore);
  });
});
```

to make it easier to read